File: Python/programmers/etc/level2/짝지어_제거하기.py
def solution(s):
    stack = []
    for i in s:
        if stack and stack[-1] == i:
            stack.pop()
        else:
            stack.append(i)

    return 1 if not stack else 0


s = "baabaa"
print(solution(s)) # 1

s = "cdcd"
print(solution(s)) # 0

File: Java/programmers/etc/level2/짝지어제거하기.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 짝지어제거하기 {
    public int solution(String s) {
        Stack<Character> stk = new Stack<>();
        
        for (char i : s.toCharArray()) {
            if (!stk.isEmpty() && stk.peek() == i) stk.pop();
            else stk.push(i);
        }
        
        return stk.isEmpty() ? 1 : 0;
    }
}
